src/modules/pivotTable/applyLegendSet.js

```diff
@@ -0,0 +1,68 @@
+import { isColorBright } from './isColorBright'
+
+const LEGEND_DISPLAY_STRATEGY_BY_DATA_ITEM = 'BY_DATA_ITEM'
+const LEGEND_DISPLAY_STRATEGY_FIXED = 'FIXED'
+
+const LEGEND_DISPLAY_STYLE_FILL = 'FILL'
+const LEGEND_DISPLAY_STYLE_TEXT = 'TEXT'
+
+const valueFitsLegend = (value, legend) =>
+    legend.startValue <= value && legend.endValue > value // TODO: Confirm inclusive/exclusive bounds
+
+const getLegendSet = engine => {
+    let legendSetId
+    switch (engine.visualization.legendDisplayStrategy) {
+        case LEGEND_DISPLAY_STRATEGY_BY_DATA_ITEM:
+            legendSetId = undefined // TODO: ByDXID LegendSet
+            //engine.rawData.metaData.items[dxId].legendSet
+            break
+        case LEGEND_DISPLAY_STRATEGY_FIXED:
+        default:
+            legendSetId = engine.visualization.legendSet
+                ? engine.visualization.legendSet.id
+                : undefined
+            break
+    }
+
+    return engine.legendSets[legendSetId]
+}
+
+const buildStyleObject = (legend, engine) => {
+    const style = {}
+    switch (engine.visualization.legendDisplayStyle) {
+        case LEGEND_DISPLAY_STYLE_TEXT:
+            style.color = legend.color
+            // TODO: Adapt background color for light text colors
+            break
+        case LEGEND_DISPLAY_STYLE_FILL:
+        default:
+            style.backgroundColor = legend.color
+            if (isColorBright(legend.color)) {
+                style.color = '#000000' // TODO: Don't hard-code color here!
```

Comment:
Do we have some colors for this in `ui-core` colors/theme?

Reply:
@cooper-joe I think this is a question for you - what should be text color be for both (a) dark and (b) light background colors in a legend-styled pivot table?  Right now the default is #000 text for light backgrounds and #FFF text for dark backgrounds, but maybe we want something slightly different?

Reply:
Another @cooper-joe question - what should happen on hover for cells with a background-color set by the legend?  Currently nothing does, but non-legend cells have a hover bg-color of [#F3F3F3](https://github.com/dhis2/analytics/blob/master/src/components/PivotTable/styles/PivotTable.style.js#L77)

Reply:
1. Light background color can use `grey900` and dark backgrounds can use `white` from ui-core.colors.

2. I don't think we should apply the standard hover color on legend cells, it would be confusing. However, trying to compute a slightly darker shade of the legend-cell-color also sounds quite intensive and is actually changing the data (modifying the legend color modifies its meaning). Therefore, let's keep it that legend cells do not have a hover state.

Reply:
👍 thanks @cooper-joe !